.github/workflows/deploy.yml

```diff
@@ -48,33 +56,39 @@ jobs:
     steps:
       - name: Check out repository
         uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # https://github.com/actions/checkout/releases/tag/v3.5.3
-      - name: Set preview in package.json
+
+      - name: Set prerelease in package.json
+        if: inputs.deploy_type == 'prerelease'
         id: set-preview
         run: |
           ./build/preview.sh
         env:
           LANGUAGE_SERVER_VERSION: ${{ github.event.inputs.langserver }}
+
       - name: Setup Node
         uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # https://github.com/actions/setup-node/releases/tag/v3.6.0
         with:
           node-version-file: '.nvmrc'
+
       - name: Install dependencies
         run: npm ci
         env:
           npm_config_arch: ${{ matrix.npm_config_arch }}
           ls_target: ${{ matrix.ls_target }}
-      - name: Package VSIX
-        run: npm run package -- --pre-release --target=${{ matrix.vsce_target }}
+
+      - name: Package PreRelease VSIX
+        if: inputs.deploy_type == 'prerelease'
+        run: npm run package -- --githubBranch $GITHUB_REF_NAME --pre-release --target=${{ matrix.vsce_target }}
+
+      - name: Package Stable VSIX
+        if: inputs.deploy_type == 'stable'
+        run: npm run package -- --target=${{ matrix.vsce_target }}
```

Comment:
I wasn't implying in my earlier comment that I necessarily _want_ us to be cutting stable releases from other branches, but rather that GHA will AFAIK always display the branch dropdown menu in the dialog with `workflow_dispatch` and so it may result in a slightly confusing UX if we don't account for that.

Maybe a minor UX issue, esp. if there's just 3 of us cutting releases and we're all aware of this - so feel free to ignore - but I wanted to clarify the risk.